hoomd/md/integrate.py

```diff
@@ -109,18 +112,48 @@ class Integrator(_DynamicIntegrator):
     - `hoomd.md.methods.npt`
     - `hoomd.md.methods.nph`
 
-    There can only be one integration mode active at a time. If there are more
-    than one ``integrate.mode_*`` commands in a hoomd script, only the most
-    recent before a given ```hoomd.run``` will take effect.
+    The classes of following modules can be used as elements in `forces`
+
+    - `hoomd.md.angle`
+    - `hoomd.md.bond`
+    - `hoomd.md.charge`
+    - `hoomd.md.dihedral`
+    - `hoomd.md.external`
+    - `hoomd.md.force`
+    - `hoomd.md.improper`
+    - `hoomd.md.pair`
+    - `hoomd.md.wall`
+    - `hoomd.md.special_pair`
+    - `hoomd.dem.pair`
+
+    The classes of the following module can be used as elements in `constraints`
+
+    - `hoomd.md.constrain`
 
     Examples::
 
-        integrate.mode_standard(dt=0.005) integrator_mode =
-        integrate.mode_standard(dt=0.001)
+        integrator = hoomd.md.Integrator(dt=0.001, methods=[nve], forces=[lj])
+        sim.operations.integrator = integrator
+
+
+    Attributes:
+        dt (float): Integrator time step size (in time units).
+
+        methods (List[hoomd.md.methods._Method]): List of integration methods.
+            Each integration method can be applied to only a specific subset of
+            particles.
+
+        forces (List[hoomd.md.force._Force]): List of forces applied to
+            the particles in the system. All the forces are summed together.
+
+        aniso (str): Whether rotational degrees of freedom are integrated.
+
+        constraints (List[hoomd.md.constrain._ConstraintForce]): List of
+            constraint forces applied to the particles in the system.
     """
 
-    def __init__(self, dt, aniso=None, forces=None, constraints=None,
-                 methods=None):
+    def __init__(self, dt, aniso='auto', forces=[], constraints=[],
+                 methods=[]):
```

Comment:
Keep `None` as the default for forces, constraints, and methods. Method defaults need to be immutable in Python otherwise they can be changed which is not desired.